packages/library/src/send-transaction.ts

```diff
@@ -0,0 +1,200 @@
+        config: Omit<
+            SendTransactionConfig &
+                Readonly<{
+                    abortSignal: AbortSignal;
+                    commitment: Commitment;
+                }>,
+            'encoding'
+        >
+    ): Promise<void> {
+        await sendAndConfirmDurableNonceTransaction({
+            ...config,
+            confirmDurableNonceTransaction,
+            rpc,
+            transaction,
+        });
+    };
+}
+
+export function createDefaultTransactionSender({ rpc, rpcSubscriptions }: RecentTransactionSenderConfig) {
+    const confirmRecentTransaction = createDefaultRecentTransactionConfirmer({
+        rpc,
+        rpcSubscriptions,
+    });
+    return async function sendTransaction(
+        transaction: SendableTransaction & ITransactionWithBlockhashLifetime,
+        config: Omit<
+            SendTransactionConfig &
+                Readonly<{
+                    abortSignal: AbortSignal;
+                    commitment: Commitment;
+                }>,
+            'encoding'
+        >
+    ): Promise<void> {
+        await sendAndConfirmTransaction({
+            ...config,
+            confirmRecentTransaction,
+            rpc,
+            transaction,
+        });
+    };
+}
+
+export async function sendAndConfirmDurableNonceTransaction({
+    abortSignal,
+    commitment,
+    confirmDurableNonceTransaction,
+    rpc,
+    transaction,
+    ...sendTransactionConfig
+}: Omit<
+    SendTransactionConfig &
+        Readonly<{
+            abortSignal: AbortSignal;
+            commitment: Commitment;
+            confirmDurableNonceTransaction: ReturnType<typeof createDefaultDurableNonceTransactionConfirmer>;
+            rpc: Rpc<SendTransactionApi>;
+            transaction: SendableTransaction & IDurableNonceTransaction;
+        }>,
+    'encoding'
+>) {
+    await sendTransaction_INTERNAL({
+        abortSignal,
+        commitment,
+        rpc,
+        sendTransactionConfig,
+        transaction,
+    });
+    await confirmDurableNonceTransaction({
+        abortSignal,
+        commitment,
+        transaction,
+    });
+}
+
+export async function sendAndConfirmTransaction({
+    abortSignal,
+    commitment,
+    confirmRecentTransaction,
+    rpc,
+    transaction,
+    ...sendTransactionConfig
+}: Omit<
+    SendTransactionConfig &
+        Readonly<{
+            abortSignal: AbortSignal;
+            commitment: Commitment;
+            confirmRecentTransaction: ReturnType<typeof createDefaultRecentTransactionConfirmer>;
+            rpc: Rpc<SendTransactionApi>;
+            transaction: SendableTransaction & ITransactionWithBlockhashLifetime;
+        }>,
+    'encoding'
+>) {
+    await sendTransaction_INTERNAL({
+        abortSignal,
+        commitment,
+        rpc,
+        sendTransactionConfig,
+        transaction,
+    });
+    await confirmRecentTransaction({
+        abortSignal,
+        commitment,
+        transaction,
+    });
```

Comment:
I made the choice _not_ to return the transaction signature from here.

Pros:
* reduces the demand on the send/confirm functions (ie. they're not responsible for vending a signature)

Cons:
* if you want it (eg. for logging) you have to get it yourself using `getSignatureFromTransaction()`, as in the test plan above

Acceptable? I can't decide.

Reply:
This seems weird to me.

1. `sendTransaction` RPC method returns the signature, but `sendAndConfirm` doesn't?
2. People are used to being vended a signature, even if it's already confirmed it's a common pattern for dApps to create links to explorers or simply display the confirmed transaction signature
3. You've already wrapped lots of useful bits, why not finish with `getSignatureFromTransaction()`?

Reply:
Agree, I'd prefer to have this return a signature like `sendTransaction` too. I think that'd actually save work too because this function has the signature already, whereas the caller would have to use `getSignatureFromTransaction` to get it.